2.0/m_antirandom.cpp

```diff
@@ -623,56 +616,43 @@ class ModuleAntiRandom : public Module
 
 		if (digits >= 5)
 		{
-			score += 5 + (digits - 5);
+			score += digits;
 			if (this->DebugMode)
-				ServerInstance->SNO->WriteGlobalSno('a', "m_antirandom: %s:MATCH digits", original_str);
+				ServerInstance->SNO->WriteGlobalSno('a', "m_antirandom: %s:MATCH digits", original_str.c_str());
 		}
 		if (vowels >= 4)
 		{
-			score += 4 + (vowels - 4);
+			score += vowels;
 			if (this->DebugMode)
-				ServerInstance->SNO->WriteGlobalSno('a', "m_antirandom: %s:MATCH vowels", original_str);
+				ServerInstance->SNO->WriteGlobalSno('a', "m_antirandom: %s:MATCH vowels", original_str.c_str());
 		}
 		if (consonants >= 4)
 		{
-			score += 4 + (consonants - 4);
+			score += consonants;
 			if (this->DebugMode)
-				ServerInstance->SNO->WriteGlobalSno('a',  "m_antirandom: %s:MATCH consonants", original_str);
+				ServerInstance->SNO->WriteGlobalSno('a',  "m_antirandom: %s:MATCH consonants", original_str.c_str());
 		}
 
 
 		/*
 		 * Now, do the triples checks. For each char in the string we're checking ...
-		 * XXX - on reading this, I wonder why we can't strcmp()/strchr() it..
 		 */
-		for (s = original_str; *s; s++)
+		if (original_str.length() >= 3) // Make sure the string is at least 3 characters long
 		{
-			/*
-			 * ..run it through each triple.
-			 */
-			for (ci = triples_txt; *ci; ci++)
+			for (i = 0; i < (original_str.length() - 2); i++)
 			{
-				// At this point, ci[0] and ci[1] point to the first two chars in the triples array.
-				if (*ci[0] == s[0] && *ci[1] == s[1] && s[2])
+				std::map<std::string, std::string>::const_iterator trip;
+				// Check whether the current and next characters are the first half of a triple, if so, check for the 3rd character in the second half
+				if ((trip = triples_map.find(original_str.substr(i, 2))) != triples_map.end() &&
```

Comment:
Assign the value to the iterator on the previous line where the variable is declared